activerecord/lib/rails/generators/active_record/migration/migration_generator.rb

```diff
@@ -69,6 +74,10 @@ def validate_file_name!
         def normalize_table_name(_table_name)
           pluralize_table_names? ? _table_name.pluralize : _table_name.singularize
         end
+
+        def primary_file?
+          shell.base.class.to_s == "Rails::Generators::MigrationGenerator"
+        end
```

Comment:
I don't understand how these `primary_file?` checks work?

Reply:
This generator will be triggered from `Rails::Generators::MigrationGenerator`, so this check works(Does this make the answer as expected...?).

Reply:
I don't understand what you mean. Can you give an example of the complete code flow?

Reply:
Perhaps it's just because I don't understand what `shell` or `shell.base` is here.